mnist_hogwild/train.py

```diff
@@ -47,16 +51,16 @@ def train_epoch(epoch, args, model, data_loader, optimizer):
                 100. * batch_idx / len(data_loader), loss.item()))
 
 
-def test_epoch(model, data_loader):
+def test_epoch(model, device, data_loader):
     model.eval()
     test_loss = 0
     correct = 0
     with torch.no_grad():
         for data, target in data_loader:
-            output = model(data)
-            test_loss += F.nll_loss(output, target, reduction='sum').item() # sum up batch loss
+            output = model(data.to(device))
+            test_loss += F.nll_loss(output, target.to(device), reduction='sum').item() # sum up batch loss
             pred = output.max(1)[1] # get the index of the max log-probability
-            correct += pred.eq(target).sum().item()
+            correct += pred.eq(target.to(device)).sum().item()
```

Comment:
I don't think you want to transfer the target to CUDA twice in short succession?